File: CPP/Problems/Integer to Roman.cpp
class Solution {
public:
    string intToRoman(int num) { 
        const int val[13] = {1000,900,500,400,100,90,50,40,10,9,5,4,1};
        const string rom[13] = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
 
        string ans = "";
        for(int i = 0 ; num >0; i++){
            while(num>=val[i]) ans+=rom[i], num-=val[i];
        }
        return ans;
    }
};

File: Python/IntegerToRoman.py
"""
https://leetcode.com/problems/integer-to-roman/
Given an integer, convert it to a roman numeral.
"""
class Solution(object):
    def intToRoman(self, num):
        """
        :type num: int
        :rtype: str
        """
        pattern = [("M",1000),
                  ("CM",900),
                  ("D",500),
                  ("CD",400),
                  ("C",100),
                  ("XC",90),
                  ("L",50),
                  ("XL",40),
                  ("X",10),
                  ("IX",9),
                  ("V",5),
                  ("IV",4),
                  ("I",1)]
        output = ""
        for roman, corresonding_number in pattern:
            output += roman * (num//corresonding_number)
            num = num % corresonding_number
        return output
